sklearn/linear_model/_sgd_fast.pyx

```diff
@@ -680,10 +675,16 @@ def _plain_sgd(cnp.ndarray[double, ndim=1, mode='c'] weights,
 
     w.reset_wscale()
 
-    return weights, intercept, average_weights, average_intercept, epoch + 1
+    return (
+        weights.base,
+        intercept,
+        None if average_weights is None else average_weights.base,
+        average_intercept,
+        epoch + 1
+    )
 
 
-cdef bint any_nonfinite(double *w, int n) nogil:
+cdef bint any_nonfinite(const double *w, int n) nogil:
```

Comment:
@jjerphan do you think that we should delay the use of `const` and only restrict to the memoryview when changing it? I know it is a good practice, but we have that many variables that are not defined as `const` (and pointers), I assume it would be best to have a dedicated PR for a specific function or file.

Reply:
This changes is necessary for removing the warning (i.e. not using the old NumPy API hence not using `cnp.ndarray` hence using memoryviews instead) while supporting readonly/memmaped data since the pointer references the buffer of a `const`-qualified memoryview. I do not think we can get our way around the modification here without casting `const`- to non-`const`-qualified variable.

But I agree generally and also think that we _must_ not change signatures to add `const`-qualifier if they aren't required for removing compilation warnings.

What do you think?

Reply:
> This changes is necessary for removing the warning

OK I missed it. I thought it was introduced in the last commit. My bad.